scripts/install

```diff
@@ -1,33 +1,81 @@
 #!/bin/bash
 
-# Exit on command errors and treat unset variables as an error
-set -eu
+#=================================================
+# GENERIC START
+#=================================================
+# IMPORT GENERIC HELPERS
+#=================================================
+
+source _common.sh
+source /usr/share/yunohost/helpers
+
+#=================================================
+# MANAGE SCRIPT FAILURE
+#=================================================
+
+ynh_clean_setup () {
+	### Remove this function if there's nothing to clean before calling the remove script.
+	true
+}
+# Exit if an error occurs during the execution of the script
+ynh_abort_if_errors
+
+#=================================================
+# RETRIEVE ARGUMENTS FROM THE MANIFEST
+#=================================================
 
-# Retrieve arguments
 app=$YNH_APP_INSTANCE_NAME
 domain=$YNH_APP_ARG_DOMAIN
-path="/"
+path_url="/"
 port=10901
 admin=$YNH_APP_ARG_ADMIN
 
-# Source app helpers and functions
-source /usr/share/yunohost/helpers
-source functions.sh
+#=================================================
+# CHECK IF THE APP CAN BE INSTALLED WITH THESE ARGS
+#=================================================
 
+### If the app uses nginx as web server (written in HTML/PHP in most cases), the final path should be "/var/www/$app".
+### If the app provides an internal web server (or uses another application server such as uwsgi), the final path should be "/opt/yunohost/$app"
+final_path=/var/www/$app
+test ! -e "$final_path" || ynh_die "This path already contains a folder"
+
+# Normalize the url path syntax
+path_url=$(ynh_normalize_url_path $path_url)
+
+# Check web path availability
+ynh_webpath_available $domain $path_url
+# Register (book) web path
+ynh_webpath_register $app $domain $path_url
+
+#=================================================
 # Check domain/path availability
+#=================================================
+
 yunohost app checkurl "${domain}${path}" -a "$app" \
     || ynh_die "Path not available: ${domain}${path}"
 
+#=================================================
 # Check port availability
+#=================================================
+
 yunohost app checkport $port
 if [[ ! $? -eq 0 ]]; then
   ynh_die "Port not available: ${port}"
+else
+  # Open port on firewall
+  yunohost firewall allow TCP $port > /dev/null 2>&1
```

Comment:
Already present down there.
I would not open a port if `ynh_user_exists` fails.

Reply:
Is this possible to have yunohost with a admin account ?

Reply:
In this case, a YunoHost user will be an admin of the Duniter instance.
If the user does not exists, it will fails, and the port will remains open.

Reply:
I did a correction. But the port can remain open as if there is an error, remove script is called, and it close port anyway.

Reply:
Ah, yes, you are right. Thanks.